public/app/features/playlist/api.ts

```diff
@@ -1,27 +1,36 @@
 import { getBackendSrv } from '@grafana/runtime';
 
-import { Playlist } from './types';
+import { Playlist, PlaylistDTO } from './types';
 import { dispatch } from '../../store/store';
 import { notifyApp } from '../../core/actions';
 import { createErrorNotification, createSuccessNotification } from '../../core/copy/appNotification';
 
 export async function createPlaylist(playlist: Playlist) {
-  await withErrorHandling(async () => await getBackendSrv().post('/api/playlists', playlist));
+  await withErrorHandling(() => getBackendSrv().post('/api/playlists', playlist));
 }
 
 export async function updatePlaylist(id: number, playlist: Playlist) {
-  await withErrorHandling(async () => await getBackendSrv().put(`/api/playlists/${id}`, playlist));
+  await withErrorHandling(() => getBackendSrv().put(`/api/playlists/${id}`, playlist));
+}
+
+export async function deletePlaylist(id: number) {
+  await withErrorHandling(() => getBackendSrv().delete(`/api/playlists/${id}`), 'Playlist deleted');
 }
 
 export async function getPlaylist(id: number): Promise<Playlist> {
   const result: Playlist = await getBackendSrv().get(`/api/playlists/${id}`);
   return result;
 }
 
-async function withErrorHandling(apiCall: () => Promise<void>) {
+export async function getAllPlaylist(query: string): Promise<PlaylistDTO[]> {
+  const result: PlaylistDTO[] = await getBackendSrv().get('/api/playlists/', query);
```

Comment:
I think the call to the api used to be. Not sure if passing just a string would work.
```suggestion
  const result: PlaylistDTO[] = await getBackendSrv().get('/api/playlists/', { query });
```

Reply:
The `getBackendSrv().get()` function accepts 3 args that are of type `string` and `any`. So I think passing just a string would work. @hugohaggmark

Reply:
Not sure if this makes sense but the backend api needs a param called `query` and they way we provide it from the frontend is to pass an object with the property needed for the backend api call, in this case a property named `query`.
https://github.com/grafana/grafana/blob/6cfb640a0b6ba59a512f05f4fe5fb0e7865da322/pkg/api/playlist.go#L31

Let me know if something is unclear.

Reply:
oh...ok.
Now it makes sense. Thanks for the clarification